src/transformers/pipelines/base.py

```diff
@@ -585,6 +595,49 @@ def predict(self, X):
             Flag indicating if the output the pipeline should happen in a binary format (i.e., pickle) or as raw text.
 """
 
+if is_torch_available():
+
+    class PipelineDataset(Dataset):
+        def __init__(self, dataset, process):
+            self.dataset = dataset
+            self.process = process
+
+        def __len__(self):
+            return self.dataset.__len__()
```

Comment:
Magic methods are not meant to be called directly.
```suggestion
            return len(self.dataset)
```